src/v/utils/named_type.h

```diff
@@ -180,6 +180,12 @@ class base_named_type<T, Tag, std::false_type> {
     type _value;
 };
 
+template<typename T, typename Tag, typename IsConstexpr>
```

Comment:
I ran an experiment with this implemented as hidden friends and the difference in compile time was insignificant.

Reply:
that's good